### PR TITLE
fix(dev): Do not export SENTRY_DSN in .envrc

### PR DESCRIPTION
**Pull request by armenzg**
We started exporting SENTRY_DSN back in #32524 in order to fix errors in lib.sh not being reported.

Having this variable exported can cause a lot of errors from a developers' development to show up as dev env errors.

Exporting inside of lib.sh does not persist in the developers' environment.

### CODE REVIEW

File: scripts/do.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 # This script is an interface to any of the methods of lib.sh
 # Call this script as "do.sh method_from_lib" to execute any function from that library
-set -eu
+set -e
```
Comment (armenzg) [inline scripts/do.sh:4]:
```suggestion
set -eu
```

Comment (armenzg) [reply scripts/do.sh:4]: This is probably unnecessary and only when I was doing tests.

File: scripts/lib.sh [hunk 2]
```diff
@@ -28,10 +28,13 @@ configure-sentry-cli() {
     # We can remove this after it's fixed
     # https://github.com/getsentry/sentry-cli/pull/1059
     export SENTRY_CLI_NO_EXIT_TRAP=${SENTRY_CLI_NO_EXIT_TRAP-0}
-    if [ -n "${SENTRY_DSN+x}" ] && [ -z "${SENTRY_DEVENV_NO_REPORT+x}" ]; then
+    if [ -z "${SENTRY_DEVENV_NO_REPORT+x}" ]; then
```
Comment (armenzg) [inline scripts/lib.sh:31]: Since we now don't export the variable.

File: scripts/lib.sh [hunk 3]
```diff
@@ -28,10 +28,13 @@ configure-sentry-cli() {
...
         if ! require sentry-cli; then
             curl -sL https://sentry.io/get-cli/ | SENTRY_CLI_VERSION=2.0.4 bash
         fi
+        # This exported variable does not persist outside of the calling script, thus, not affecting other
+        # parts of the system
+        export SENTRY_DSN="https://9bdb053cb8274ea69231834d1edeec4c@o1.ingest.sentry.io/5723503"
```
Comment (armenzg) [inline scripts/lib.sh:37]: Once the script that calls configure-sentry-cli ends, the variable is lost and the potential impact contained.

**Comment by armenzg** [comment]
@asottile-sentry @Zylphrex would you mind testing that direnv works for you with this PR? (since you two were affected on #33826)

**Review by asottile-sentry — APPROVED** [review]
wfm

**Review by flub — APPROVED** [review]

**Pull request merged**